RevenueCatUI/CustomerCenter/Views/PromotionalOfferView.swift

```diff
@@ -0,0 +1,177 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PromotionalOfferView.swift
+//
+//
+//  Created by Cesar de la Vega on 17/6/24.
+//
+
+import RevenueCat
+import StoreKit
+import SwiftUI
+
+#if os(iOS)
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@available(visionOS, unavailable)
+struct PromotionalOfferView: View {
+
+    @StateObject
+    private var viewModel: PromotionalOfferViewModel
+    @Environment(\.dismiss)
+    private var dismiss
+
+    init(promotionalOffer: PromotionalOffer,
+         product: StoreProduct,
+         promoOfferDetails: CustomerCenterConfigData.HelpPath.PromotionalOffer,
+         localization: CustomerCenterConfigData.Localization) {
+        let promotionalOfferData = PromotionalOfferData(promotionalOffer: promotionalOffer,
+                                                        product: product,
+                                                        promoOfferDetails: promoOfferDetails)
+        let viewModel = PromotionalOfferViewModel(promotionalOfferData: promotionalOfferData,
+                                                  localization: localization)
+        self._viewModel = StateObject(wrappedValue: viewModel)
+    }
+
+    var body: some View {
+        VStack {
+            if let details = self.viewModel.promotionalOfferData?.promoOfferDetails,
+               let localization = self.viewModel.localization {
+                Text(details.title)
+                    .font(.title)
+                    .padding()
+
+                Text(details.subtitle)
+                    .font(.title3)
+                    .padding()
+
+                Spacer()
+
+                PromoOfferButtonView(viewModel: viewModel)
+
+                let title = localization.commonLocalizedString(for: .noThanks) ?? "No, thanks"
```

Comment:
Should we move the defaults to the `Localization` struct? That way we centralize it there.

Reply:
Also, maybe rename to `dismissButtonTitle`?